provider/mem/mem.go

```diff
@@ -174,11 +191,13 @@ func (a *Alerts) GetPending() provider.AlertIterator {
 		ch   = make(chan *types.Alert, alertChannelLength)
 		done = make(chan struct{})
 	)
+	a.mtx.RLock()
+	defer a.mtx.RUnlock()
+	alerts := a.alerts.List()
```

Comment:
I think the important change here is that moving the alerts out of the goroutine means `alerts` contains the set of alerts at the time `Pending()` was called, not the set of alerts at the time the goroutine was scheduled. This can be different if a `Put` or GC operation interleaves a call to `Pending()` and the start of the goroutine.